File: src/Calculator/Converters/VisibilityNegationConverter.cpp
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

#include "pch.h"
#include "VisibilityNegationConverter.h"

using namespace Platform;
using namespace Windows::Foundation;
using namespace Windows::UI::Xaml;
using namespace Windows::UI::Xaml::Interop;

namespace CalculatorApp
{
    namespace Common
    {
        Object ^ VisibilityNegationConverter::Convert(Object ^ value, TypeName /*targetType*/, Object ^ /*parameter*/, String ^ /*language*/)
        {
            auto boxedVisibility = dynamic_cast<Box<Visibility> ^>(value);
            if (boxedVisibility != nullptr && boxedVisibility->Value == Visibility::Collapsed)
            {
                return Visibility::Visible;
            }
            return Visibility::Collapsed;
        }

        Object ^ VisibilityNegationConverter::ConvertBack(Object ^ value, TypeName targetType, Object ^ parameter, String ^ language)
        {
            return Convert(value, targetType, parameter, language);
        }
    }
}

File: src/Calculator/Converters/VisibilityNegationConverter.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using System;

using Windows.UI.Xaml;

namespace CalculatorApp
{
    namespace Common
    {
        /// <summary>
        /// Value converter that translates Visible to Collapsed and vice versa
        /// </summary>
        [Windows.Foundation.Metadata.WebHostHidden]
        public sealed class VisibilityNegationConverter : Windows.UI.Xaml.Data.IValueConverter
        {
            public object Convert(object value, Type targetType, object parameter, string language)
            {
                if (value is Visibility boxedVisibility && boxedVisibility == Visibility.Collapsed)
                {
                    return Visibility.Visible;
                }
                return Visibility.Collapsed;
            }

            public object ConvertBack(object value, Type targetType, object parameter, string language)
            {
                return Convert(value, targetType, parameter, language);
            }
        }
    }
}
